File: LeetCode/Python/two-sum.py
# https://leetcode.com/problems/two-sum/
# Reprak11

from typing import List

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dictio = {}
        for i,j in enumerate(nums):
            diff = target - j
            if diff in dictio:
                return[dictio[diff],i]
            dictio[j]=i
        return 

dato1= input()[1:-1]
nums=list(map(int,dato1.split(",")))

target=int(input())

print(Solution.twoSum(Solution,nums,target))

File: LeetCode/Go/twoSum.go
// https://leetcode.com/problems/two-sum/
// Reprak11
func isValueInArray(nums []int, currentValue int, currentPos int) (pos2 int, found bool) {
	for pos, value := range nums {
		if (currentPos != pos) && (value == currentValue) {
			return pos, true
		}
	}
	return -1, false
}

func twoSum(nums []int, target int) []int {
	isThere := false
	secondPos := -1
	for pos, value := range nums {
		secondPos, isThere = isValueInArray(nums, target-value, pos)
		if isThere == true {
			return []int{pos, secondPos}
		}
	}
	return []int{-1, -1}
}
